packages/app-explorer/src/Events.tsx

```diff
@@ -54,35 +44,38 @@ class Events extends React.PureComponent<Props> {
       return null;
     }
 
+    const hash = xxhashAsHex(stringToU8a(JSON.stringify(record)));
+
     return (
-      <article
-        className='explorer--Container'
-        key={index}
+      <div
+        className={eventClassName}
+        key={hash}
```

Comment:
Instead of attaching the hash here, rather create a unique id as it gets added to the array - that way when we do have overlapping events, the rendering will still point to the right items in the queue - and we won't get duplicate key errors. (Have not seen them, but it certainly can happen now since events, system with no nonce, can repeat)